### DISCUSSION TITLE
The difference between entrySpan and LocalSpan

### DISCUSSION THREAD

**Discussion by willardwang** [apache/skywalking#11159] [General]
Hello, I have a problem. When I create a logical endpoint using localSpan, I find that it does not display trace. If I use entrySpan, it will display trace.

**Comment by wu-sheng** [comment]
logical span is for statistics, not for trace query. They are irrelevant.

**Reply by willardwang** [reply]
I would like to ask if a Spring MVC request will invoke many service methods, and I want to intercept each service method and create an entry span. Is this allowed?

**Reply by wu-sheng** [reply]
> EntrySpan EntrySpan represents a service provider, which is also the endpoint on the server end. As an APM system, SkyWalking targets the application servers. Therefore, almost all the services and MQ-consumers are EntrySpans.

Ref https://skywalking.apache.org/docs/main/next/en/api/trace-data-protocol-v3/#trace-data-protocol

You could search and find the definition easily.

**Reply by willardwang** [reply]
Okay, I've got a BUG, and I don't know if it's a BUG, but look at this again
Click on the endpoint to see his statistics display empty
Refer to the first picture
<img width="1592" alt="image" src="https://github.com/apache/skywalking/assets/3304388/2cef489d-a8ec-42f8-97de-9021345b4837">
This problem occurs when I click on this endpoint
<img width="1615" alt="image" src="https://github.com/apache/skywalking/assets/3304388/ff717300-9e85-4de5-aace-65671952b8e2">

The endpoint of the selector is blank

**Reply by wu-sheng** [reply]
We didn't test Chinese text. There may be an encoding issue. Upstream always considers English text only.

**Reply by willardwang** [reply]
English this will also fail
<img width="1653" alt="image" src="https://github.com/apache/skywalking/assets/3304388/cc6e74bd-8607-4650-87a0-dad287f977a1">
It will be the same 
<img width="1649" alt="image" src="https://github.com/apache/skywalking/assets/3304388/df0bbb0b-1f0c-463b-a30e-a0bff91c4b32">

I found that it has nothing to do with Chinese, when I click on this endpoint to jump to the page, the page will make a request to get all the logical endpoints
```
{"query":"query queryEndpoints($serviceId: ID!, $keyword: String!) {\n  pods: findEndpoint(serviceId: $serviceId, keyword: $keyword, limit: 20) {\n    id\n    value: name\n     label: name\n    }\n}","variables":{"serviceId":"YXdzNzpsaXVmYXRhb0FXUzctMzFBNw==.1","duration":{"start":"2023-08-01 0858","end":"2023-08-01 0928","step":"MINUTE"},"keyword":""}}
```
It can be found that the keyword is empty, which causes the problem of picture 2  ，

A steady recurrence of this problem is to have more than 20 endpoints, then search for endpoints that are not displayed on the current endpoint page and click，The endpoint selector's page component appears empty，As shown in picture 3
 
<img width="1638" alt="image" src="https://github.com/apache/skywalking/assets/3304388/ce8b498c-2d6a-4e19-b8de-6faeacaf0b8c">

**Comment by willardwang** [comment]
<img width="1616" alt="image" src="https://github.com/apache/skywalking/assets/3304388/f2194707-e7b5-4d4c-bec0-5a2c72de8bd5">

**Comment by wu-sheng** [comment]
Search and jump of endpoint works on the official demo env, [demo.skywalking.a.o.](https://skywalking.apache.org/#demo)

I can't be sure about your case.

**Reply by wu-sheng** [reply]
Are you using Safari?
